capstone-ui/src/Components/CandidateView/CandidateView.jsx

```diff
@@ -1,15 +1,89 @@
-import * as React from "react"
+import React, { useContext } from "react"
+import { Link } from 'react-router-dom';
+import { UserContext } from "../../UserContext.js";
 
 export default function CandidateView({ candidateName }) {
 
+    const { user, setUser } = useContext(UserContext);
+
+    const handleFollow = async (e) => {
+        e.preventDefault();
+      
+        try {
+          const response = await fetch("http://localhost:3001/follow", {
+            method: "POST",
+            headers: {
+              "Content-Type": "application/json",
+            },
+            body: JSON.stringify({
+              username: user.username,
+              candidateName,
+            }),
+          });
+      
+          if (response.ok) {
+            setUser((prevUser) => {
+              return {
+                ...prevUser,
+                following: [...prevUser.following, candidateName],
+              };
+            });
+          } else {
+            console.error("Failed to follow candidate:", response.status, response.statusText);
+          }
+        } catch (error) {
+            console.error("Error following candidate:", error);
+        }
+      };
+    
+      const handleUnfollow = async (e) => {
+        e.preventDefault();
+    
+        try {
+          const response = await fetch("http://localhost:3001/unfollow", {
```

Comment:
Is there a more generic way to represent the endpoint. I get the feeling that this is hardcoding?

Reply:
Not that I now of, it's the way we learned it from the codepath course. This is the way I've done all the requests, I don't know a different way.